src/bundle/Resources/public/js/scripts/core/multilevel.popup.menu.js

```diff
@@ -0,0 +1,385 @@
+(function (global, doc, ibexa, Popper) {
+    class MultilevelPopupMenu {
+        constructor(config) {
+            this.container = config.container;
+            this.triggerElement = config.triggerElement;
+            this.referenceElement = config.referenceElement ?? this.triggerElement;
+            this.onTopBranchClosed = config.onTopBranchClosed ?? (() => {});
+            this.onTopBranchOpened = config.onTopBranchOpened ?? (() => {});
+            this.processBranchOnInitAfter = config.processBranchOnInitAfter ?? (() => {});
+            this.processItemOnInitAfter = config.processItemOnInitAfter ?? (() => {});
+            this.initialBranchPlacement = config.initialBranchPlacement ?? 'right-start';
+            this.initialBranchFallbackPlacements = config.initialBranchFallbackPlacements ?? ['right-end', 'left-start', 'left-end'];
+
+            this.hoveredItemsBranches = new Set();
+            this.hoveredBranches = new Set();
+
+            this.handleClickOutside = this.handleClickOutside.bind(this);
+            this.handleItemWithSubitemsClick = this.handleItemWithSubitemsClick.bind(this);
+
+            doc.addEventListener('click', this.handleClickOutside, false);
+        }
+
+        init() {
+            const topBranch = this.container.querySelector('.ibexa-multilevel-popup-menu__branch');
+
+            if (!topBranch) {
+                return;
+            }
+
+            const itemsWithSubitems = this.container.querySelectorAll('.ibexa-popup-menu__item--has-subitems');
+
+            this.initBranch(
+                this.triggerElement,
+                topBranch,
+                this.referenceElement,
+                this.initialBranchPlacement,
+                this.initialBranchFallbackPlacements,
+                this.processBranchOnInitAfter,
+                this.processItemOnInitAfter,
+            );
+            this.triggerElement.branchElement = topBranch;
+
+            itemsWithSubitems.forEach((itemElement) => {
+                const branchElement = itemElement.querySelector(':scope > .ibexa-multilevel-popup-menu__branch');
+                const parentBranchElement = itemElement.closest('.ibexa-popup-menu');
+
+                this.initBranch(
+                    itemElement,
+                    branchElement,
+                    undefined,
+                    undefined,
+                    undefined,
+                    this.processBranchOnInitAfter,
+                    this.processItemOnInitAfter,
+                );
+
+                itemElement.branchElement = branchElement;
+                branchElement.itemElement = itemElement;
+                branchElement.parentBranchElement = parentBranchElement;
+            });
+        }
+
+        initBranch(
```

Comment:
Personally for me it's to many arguments, I would change it to object, as, for example - in line 47 I was totally lost what these three undefined meant